jax/_src/numpy/lax_numpy.py

```diff
@@ -5222,70 +5222,82 @@ def argwhere(a, *, size=None, fill_value=None):
-def argmax(a, axis: Optional[int] = None, out=None):
-  return _argmax(a, None if axis is None else operator.index(axis))
+def argmax(a, axis: Optional[int] = None, out=None, keepdims=None):
+  return _argmax(a, None if axis is None else operator.index(axis), keepdims=keepdims)
 
-@partial(jit, static_argnames=('axis',), inline=True)
-def _argmax(a, axis: Optional[int] = None, out=None):
+@partial(jit, static_argnames=('axis', 'keepdims'), inline=True)
+def _argmax(a, axis: Optional[int] = None, out=None, keepdims=None):
   _check_arraylike("argmax", a)
   if out is not None:
     raise NotImplementedError("The 'out' argument to jnp.argmax is not supported.")
   if axis is None:
+    dims = list(range(ndim(a)))
     a = ravel(a)
     axis = 0
+  else:
+    dims = [axis]
   if a.shape[axis] == 0:
     raise ValueError("attempt to get argmax of an empty sequence")
-  return lax.argmax(a, _canonicalize_axis(axis, a.ndim), dtypes.canonicalize_dtype(int_))
+  result = lax.argmax(a, _canonicalize_axis(axis, a.ndim), dtypes.canonicalize_dtype(int_))
+  return expand_dims(result, dims) if keepdims else result
 
 @_wraps(np.argmin, skip_params=['out'])
-def argmin(a, axis: Optional[int] = None, out=None):
-  return _argmin(a, None if axis is None else operator.index(axis))
+def argmin(a, axis: Optional[int] = None, out=None, keepdims=None):
+  return _argmin(a, None if axis is None else operator.index(axis), keepdims=keepdims)
 
-@partial(jit, static_argnames=('axis',), inline=True)
-def _argmin(a, axis: Optional[int] = None, out=None):
+@partial(jit, static_argnames=('axis', 'keepdims'), inline=True)
+def _argmin(a, axis: Optional[int] = None, out=None, keepdims=None):
   _check_arraylike("argmin", a)
   if out is not None:
     raise NotImplementedError("The 'out' argument to jnp.argmin is not supported.")
   if axis is None:
+    dims = list(range(ndim(a)))
     a = ravel(a)
     axis = 0
+  else:
+    dims = [axis]
   if a.shape[axis] == 0:
     raise ValueError("attempt to get argmin of an empty sequence")
-  return lax.argmin(a, _canonicalize_axis(axis, a.ndim), dtypes.canonicalize_dtype(int_))
+  result = lax.argmin(a, _canonicalize_axis(axis, a.ndim), dtypes.canonicalize_dtype(int_))
+  return expand_dims(result, dims) if keepdims else result
 
 
 _NANARG_DOC = """\
 Warning: jax.numpy.arg{} returns -1 for all-NaN slices and does not raise
 an error.
 """
 
-@_wraps(np.nanargmax, lax_description=_NANARG_DOC.format("max"))
-def nanargmax(a, axis: Optional[int] = None):
-  return _nanargmax(a, None if axis is None else operator.index(axis))
+@_wraps(np.nanargmax, lax_description=_NANARG_DOC.format("max"), skip_params=['out'])
+def nanargmax(a, axis: Optional[int] = None, out : Any = None, keepdims : Optional[bool] = None):
+  if out is not None:
+    raise NotImplementedError("The 'out' argument to jnp.nanargmax is not supported.")
+  return _nanargmax(a, None if axis is None else operator.index(axis), keepdims=bool(keepdims))
```

Comment:
OOC, why is this `keepdims` wrapped in `bool` and not the other ones?
Not super important, but wrapping them in `bool` seems to be a good thing since it saves a jit when the outer function is called with `f(..., keepdims=False)`.

Reply:
I added these to appease mypy... the other private methods are not annotated so it wasn't flagged.

Reply:
I wrapped the others in bool as well!